hub/api/dataset.py

```diff
@@ -41,18 +71,26 @@ def __getitem__(self, item: Union[slice, str, int]):
             if item not in self.tensors:
                 raise TensorNotFoundError(item, self.path)
             else:
-                return self.tensors[item]
+                return self.tensors[item][self.slice]
         elif isinstance(item, slice):
             new_slice = merge_slices(self.slice, item)
-            return Dataset(self.path, self.mode, new_slice)
+            return Dataset(self.path, self.mode, new_slice, self.provider)
         else:
             raise InvalidKeyTypeError(item)
 
     def __setitem__(self, item: Union[slice, str], value):
         if isinstance(item, str):
             if isinstance(value, np.ndarray):
-                # TODO: write data and create tensor
-                # self.tensors[item] = Tensor(...)
+                write_array(
+                    value,
+                    item,
+                    storage=self.provider,
+                    batched=True,
+                )
+                ds_meta = read_dataset_meta(self.provider)
+                ds_meta["tensors"].append(item)
+                write_dataset_meta(self.provider, ds_meta)
+                self.tensors[item] = Tensor(item, self.provider)
```

Comment:
it won't let me comment but in the method `Dataset.from_path` instead of just returning `None`, could you have it `raise NotImplementedError`?